macOSUpgrade.sh

```diff
@@ -153,10 +153,37 @@ osinstallersetupdDaemonSettingsFilePath="/Library/LaunchDaemons/com.jamfps.clean
 ##Launch agent settings for filevault authenticated reboots
 osinstallersetupdAgentSettingsFilePath="/Library/LaunchAgents/com.apple.install.osinstallersetupd.plist"
 
+##Amount of time (in seconds) to allow a user to connect to AC power before moving on
+acPowerWaitTimer="120"
+
+##Icon to display during the AC Power warning
+warnIcon="/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/AlertCautionIcon.icns"
+
+##Icon to display when errors are found
+errorIcon="/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/AlertStopIcon.icns"
+
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 # FUNCTIONS
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
+wait_for_ac_power() {
+    # Loop for 120 seconds until either AC Power is detected or the timer is up
+    sysRequirementErrors=()
+    /bin/echo "Waiting for AC power..."
+    while [[ "$acPowerWaitTimer" -gt "0" ]]; do
+        if /usr/bin/pmset -g ps | grep "AC Power" > /dev/null ; then
+            /bin/echo "Power Check: OK - AC Power Detected"
+            ps -p "$jamfHelperPowerPID" > /dev/null && kill "$jamfHelperPowerPID"; wait "$jamfHelperPowerPID" 2>/dev/null
```

Comment:
use full path for `ps`, `kill` and `wait`.

Reply:
Interestingly, adding full path to the wait command produces terminal output, whereas leaving wait without the full path produces no terminal output. The others have been resolved.

Reply:
It's interesting.

```
$ cat /usr/bin/wait 
#!/bin/sh
# $FreeBSD: src/usr.bin/alias/generic.sh,v 1.2 2005/10/24 22:32:19 cperciva Exp $
# This file is in the public domain.
builtin `echo ${0##*/} | tr \[:upper:] \[:lower:]` ${1+"$@"}
```

When use `/usr/bin/wait`, it will exec built-in `wait` in another shell. 
I asked google, the `/usr/bin/wait` seems to be  for compatibility for POSIX. 

Well, it's ok for use built-in `wait`.